src/world/GameClassic/Handlers/CharacterHandler.cpp

```diff
@@ -0,0 +1,211 @@
+/*
+Copyright (c) 2014-2017 AscEmu Team <http://www.ascemu.org/>
+This file is released under the MIT license. See README-MIT for more information.
+*/
+
+#include "StdAfx.h"
+#include "git_version.h"
+#include "AuthCodes.h"
+#include "Management/WordFilter.h"
+#include "Management/ArenaTeam.h"
+#include "Management/Battleground/Battleground.h"
+#include "Server/LogonCommClient/LogonCommHandler.h"
+#include "Storage/MySQLDataStore.hpp"
+#include "Units/Players/PlayerClasses.hpp"
+#include "Server/MainServerDefines.h"
+#include "Config/Config.h"
+#include "Map/MapMgr.h"
+#include "Map/WorldCreator.h"
+
+void WorldSession::CharacterEnumProc(QueryResult* result)
+{
+    struct PlayerItem
+    {
+        uint32_t displayId;
+        uint8_t inventoryType;
+        uint32_t enchantmentId;
+    };
+
+    PlayerItem player_items[20];
+
+    struct CharEnumData
+    {
+        uint64_t guid;
+        uint8_t level;
+        uint8_t race;
+        uint8_t Class;
+        uint8_t gender;
+        uint32_t bytes;
+        uint32_t bytes2;
+        std::string name;
+        float x;
+        float y;
+        float z;
+        uint32_t mapId;
+        uint32_t zoneId;
+        uint32_t banned;
+
+        uint32_t deathState;
+        uint32_t loginFlags;
+        uint32_t flags;
+        uint32_t guildId;
+
+    };
+
+    _side = -1;
+
+    uint32_t numchar = 0;
+    uint8_t char_real_count = 0;
+
+    if (result)
+        numchar = result->GetRowCount();
+
+    uint32_t start_time = getMSTime();
+
+    WorldPacket data(SMSG_CHAR_ENUM, 1 + numchar * 200);
+    data << uint8_t(char_real_count);
+
+    if (result)
+    {
+        do
+        {
+            Field* fields = result->Fetch();
+
+            CharEnumData charEnum;
+
+            charEnum.guid = fields[0].GetUInt64();
+            charEnum.level = fields[1].GetUInt8();
+            charEnum.race = fields[2].GetUInt8();
+            charEnum.Class = fields[3].GetUInt8();
+            if (charEnum.Class == DEATHKNIGHT)
```

Comment:
We shouldn't remove it.... if someone creates a tbc server with a old char db we should check everything which is not correct. Otherwise if you send a dk char to an classic client the client would crash or ignore all other packets.
So We should check for invalid race/class for all version (cata is the highest version, so the packet building for wotlk should exclude worgen goblin and so on.....